lib/ansible/modules/system/user.py

```diff
@@ -514,7 +526,13 @@ def modify_user_usermod(self):
 
         if self.expires:
             cmd.append('--expiredate')
-            cmd.append(time.strftime(self.DATE_FORMAT, self.expires))
+            try:
+                if self.clearexpires < 0:
+                    cmd.append('')
```

Comment:
hi f3rdy, :)
according to man page of usermod:  _An empty EXPIRE_DATE argument will disable the expiration of the account._

```
knoedel@section61:/mnt/ramdisk> cd ansible; git checkout issue_20096; git remote -v; git branch --list; cd -
Switched to branch 'issue_20096'
Your branch is up-to-date with 'origin/issue_20096'.
origin  https://github.com/StefanKaerst/ansible.git (fetch)
origin  https://github.com/StefanKaerst/ansible.git (push)
  devel
* issue_20096
/mnt/ramdisk
knoedel@section61:/mnt/ramdisk> sudo ansible/hacking/test-module -m ansible/lib/ansible/modules/system/user.py -a "name=ansible expires=-1" 
* including generated source, if any, saving to: /root/.ansible_module_generated
* ansiballz module detected; extracted module source to: /root/debug_dir
***********************************
RAW OUTPUT

{"comment": "tralala", "shell": "/bin/bash", "group": 100, "name": "ansible", "changed": true, "state": "present", "invocation": {"module_args": {"comment": null, "ssh_key_bits": 0, "update_password": "always", "non_unique": false, "force": false, "skeleton": null, "ssh_key_passphrase": null, "createhome": true, "uid": null, "home": null, "append": false, "ssh_key_type": "rsa", "ssh_key_comment": "ansible-generated on section61", "group": null, "system": false, "state": "present", "shell": null, "expires": -1.0, "ssh_key_file": null, "groups": null, "move_home": false, "password": null, "name": "ansible", "seuser": null, "remove": false, "login_class": null, "generate_ssh_key": null}}, "home": "/home/ansible", "move_home": false, "append": false, "uid": 1003}


***********************************
PARSED OUTPUT
{
    "append": false, 
    "changed": true, 
    "comment": "tralala", 
    "group": 100, 
    "home": "/home/ansible", 
    "invocation": {
        "module_args": {
            "append": false, 
            "comment": null, 
            "createhome": true, 
            "expires": -1.0, 
            "force": false, 
            "generate_ssh_key": null, 
            "group": null, 
            "groups": null, 
            "home": null, 
            "login_class": null, 
            "move_home": false, 
            "name": "ansible", 
            "non_unique": false, 
            "password": null, 
            "remove": false, 
            "seuser": null, 
            "shell": null, 
            "skeleton": null, 
            "ssh_key_bits": 0, 
            "ssh_key_comment": "ansible-generated on section61", 
            "ssh_key_file": null, 
            "ssh_key_passphrase": null, 
            "ssh_key_type": "rsa", 
            "state": "present", 
            "system": false, 
            "uid": null, 
            "update_password": "always"
        }
    }, 
    "move_home": false, 
    "name": "ansible", 
    "shell": "/bin/bash", 
    "state": "present", 
    "uid": 1003
}
knoedel@section61:/mnt/ramdisk> sudo chage -l ansible
Last password change                                    : Jun 02, 2017
Password expires                                        : never
Password inactive                                       : never
Account expires                                         : never
Minimum number of days between password change          : 0
Maximum number of days between password change          : 99999
Number of days of warning before password expires       : 7
knoedel@section61:/mnt/ramdisk> 

```
btw. the user module always says "Changed: True" = issue #13235. I'm working on that too.

hth